### PR TITLE
Redact markdown headers

### PR DESCRIPTION
**Pull request by bethanyaconnor**
We would like to redact headers for Code.org markdown files like we do for hour of code markdown files. This PR is largely based on the logic we already have a moves some functions to a shared location.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/sync-in.rb [hunk 1]
```diff
@@ -297,4 +298,12 @@ def redact_block_content
   RedactRestoreUtils.redact(source, source, ['blockfield'], 'txt')
 end
 
+def sanitize_markdown_headers
```
Comment (bethanyaconnor) [inline bin/i18n/sync-in.rb:301]: We've talked in the past about cleaning up the way we pull in Markdown files (ie moving out of the in.sh script). Now that we have to explicitly touch the markdown files in the in step I could move that logic here (probably in a future PR). Thoughts?

File: bin/i18n/sync-out.rb [hunk 2]
```diff
@@ -268,4 +269,25 @@ def rebuild_blockly_js_files
   end
 end
 
+# In the sync in, we slice the YAML headers of the files we upload to crowdin
+# down to just the part we want to translate (ie, the title). Here, we
+# reinflate the header with all the values from the source file.
+def restore_markdown_headers
```
Comment (Hamms) [inline bin/i18n/sync-out.rb:275]: has this been moved from somewhere, or just copied?

File: bin/i18n/sync-in.rb (hunk 1 above)
Comment (Hamms) [reply bin/i18n/sync-in.rb:301]: I'm all about that!

File: bin/i18n/sync-out.rb (hunk 2 above)
Comment (bethanyaconnor) [reply bin/i18n/sync-out.rb:275]: Adapted from here: https://github.com/code-dot-org/code-dot-org/blob/5eb122c96bca660c9b8fd430bdf8c1e27eac7aac/bin/i18n/hoc_sync_utils.rb#L108

I thought about trying to combine them but there's just enough differences that they would need some refactoring, plus most of the calls are to shared functions anyway.

**Review by Hamms — APPROVED** [review]

**Pull request merged**